src/components/text-editor/text-editor.test.js

```diff
@@ -0,0 +1,354 @@
+const longText =
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.";
+const longTextAssert =
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore ";
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+context("Test for TextEditor component", () => {
+  describe("check functionality for TextEditor component", () => {
+    it("should check the counter works properly", () => {
+      CypressMountWithProviders(<TextEditorCustom />);
+
+      textEditorInput().clear().type(textForInput);
+      textEditorCounter().should("have.text", 2982);
+    });
+
+    it.each([["bold"], ["italic"]])(
+      "should render text using %s style",
+      (buttonType) => {
+        CypressMountWithProviders(<TextEditorCustom />);
+
+        textEditorToolbar(buttonType).click();
+        textEditorInput().clear().type(textForInput);
+        if (buttonType === "bold") {
+          innerText().should("have.css", "font-weight", "700");
+        } else if (buttonType === "italic") {
+          innerText().should("have.css", "font-style", "italic");
+        }
+        textEditorToolbar(buttonType).should(
+          "have.css",
+          "background-color",
+          "rgb(204, 214, 219)"
+        );
+      }
+    );
+
+    it.each([["bullet-list"], ["number-list"]])(
+      "should render text in %s style",
+      (buttonType) => {
+        CypressMountWithProviders(<TextEditorCustom />);
+
+        textEditorToolbar(buttonType).click();
+        textEditorInput().clear().type("Testing{Enter}is{Enter}awesome");
+        if (buttonType === "bullet-list") {
+          innerTextList("ul", positionOfElement("second")).should(
+            "have.text",
+            "Testing"
+          );
+          innerTextList("ul", positionOfElement("third")).should(
+            "have.text",
+            "is"
+          );
+          innerTextList("ul", positionOfElement("fourth")).should(
+            "have.text",
+            "awesome"
+          );
+        } else if (buttonType === "bullet-list-numbers") {
+          innerTextList("ol", positionOfElement("second")).should(
+            "have.text",
+            "Testing"
+          );
+          innerTextList("ol", positionOfElement("third")).should(
+            "have.text",
+            "is"
+          );
+          innerTextList("ol", positionOfElement("fourth")).should(
+            "have.text",
+            "awesome"
+          );
+        }
+        textEditorToolbar(buttonType).should(
+          "have.css",
+          "background-color",
+          "rgb(204, 214, 219)"
+        );
+      }
+    );
+
+    it("should focus bold button using Tab keyboard", () => {
+      CypressMountWithProviders(<TextEditorCustom />);
+      textEditorInput().focus();
+      cy.focused().tab();
+      textEditorToolbar("bold").should("be.focused");
+    });
+
+    it.each([
```

Comment:
This is the same test as the one above, I think you can delete the one on line 166